src/Token/TokenBase.tsx

```diff
@@ -0,0 +1,92 @@
+import {KeyboardEvent} from 'react'
+import styled, {css} from 'styled-components'
+import {variant} from 'styled-system'
+import {get} from '../constants'
+
+export type TokenSizeKeys = 'sm' | 'md' | 'lg' | 'xl'
+
+export const tokenSizes: Record<TokenSizeKeys, number> = {
+  sm: 16,
+  md: 20,
+  lg: 24,
+  xl: 32
+}
+
+export const defaultTokenSize = 'md'
+
+export interface TokenBaseProps
+  extends Omit<React.HTMLProps<HTMLSpanElement | HTMLButtonElement | HTMLAnchorElement>, 'size' | 'id'> {
+  as?: 'button' | 'a' | 'span'
+  handleRemove?: () => void
+  isSelected?: boolean
+  tabIndex?: number
+  text: string
+  id?: number | string
+  variant?: TokenSizeKeys
+}
+
+export const isTokenInteractive = ({as = 'span', onClick, onFocus, tabIndex = -1}: TokenBaseProps) =>
+  Boolean(onFocus || onClick || tabIndex > -1 || ['a', 'button'].includes(as))
+
+const variants = variant<
+  {fontSize: number; height: string; gap: number; paddingLeft: any; paddingRight: number},
+  TokenSizeKeys
+>({
+  variants: {
+    sm: {
+      fontSize: 0,
+      gap: 1,
+      height: `${tokenSizes.sm}px`,
```

Comment:
Is there a reason we don't store the token sizes in pixel values so we don't have to do this string manipulation?

```js
const tokenSizes = {
  small: '16px'
  ...
}
```

Reply:
No reason. I agree with you - we should store the values with a unit.